lib/backends/sdc/build.js

```diff
@@ -891,8 +894,266 @@ function runBuildCommand(opts, callback) {
 }
 
 
+/**
+ * Commit (create) an image from a given container.
+ *
+ * @param {Object} opts
+ * @param {String} opts.author - Author name for the image
+ * @param {Array} opts.changes - Docker image changes to make.
+ * @param {String} opts.comment - Comment notes for the image.
+ * @param {Object} opts.config - Image config object
+ * @param {Boolean} opts.pause - common.boolFromQueryParam(req.query.pause),
+ * @param {Object} opts.req - Request instance.
+ * @param {String} opts.tag - repoAndTag
+ *
+ * @param callback {Function} `function (err, imageId)`
+ */
+function commitImage(req, opts, callback) {
+    assert.object(req, 'req');
+    assert.object(opts, 'opts');
+    assert.func(callback, 'callback');
+    assert.optionalString(opts.author, 'opts.author');
+    assert.optionalArrayOfString(opts.changes, 'opts.changes');
+    assert.optionalString(opts.comment, 'opts.comment');
+    assert.optionalObject(opts.config, 'opts.config');
+    assert.optionalBool(opts.pause, 'opts.pause');
+    assert.optionalString(opts.tag, 'opts.tag');
+
+    var vm = req.vm;
+    assert.object(vm, 'vm');  // Should have been added by the backend.
+
+    var baseImg;
+    var baseImageId = vm.internal_metadata['docker:imageid'];
+    var cleanupTimeoutId = -1;
+    var commitFinishedSuccessfully = false;
+    var finalImageId = null;  // Final docker image id for resulting image.
+    var log = req.log;
+    var previousErr;
+    var rat;
+    var req_id = req.getId();
+    var scratchImage;
+    var socketOpts;
+
+    // There is the potential to run the callback twice, so make sure that never
+    // happens.
+    callback = once(callback);
+
+    try {
+        rat = drc.parseRepoAndTag(opts.tag || '');
+    } catch (e) {
+        callback(new errors.DockerError(e, e.message));
+        return;
+    }
+
+    vasync.pipeline({ funcs: [
+        getScratchImage,
+        findContainerImage,
+        startCnapiDockerCommit,
+        runCnapiDockerCommit
+    ]}, commitCleanup);
+
+
+    function getScratchImage(_, cb) {
+        req.app.backend.getScratchImage(req,
+        function _getScratchImageCb(err, sImg) {
+            if (err) {
+                log.error(err, 'getScratchImage failure');
+                cb(err);
+                return;
+            }
+            scratchImage = sImg;
+            //log.debug('scratchImage: ', scratchImage);
```

Comment:
I found it spammy - I will drop that line (rather than comment it out). It doesn't give much additional information for docker commit (as the only place it's used is a reference check), and the contents are static (already defined in the *etc/scratch_image.json* file).